cosmwasm/packages/wasmi-runtime/src/wasm/errors.rs

```diff
@@ -11,19 +11,34 @@ pub enum WasmEngineError {
     OutOfGas,
     EncryptionError,
     DecryptionError,
-    DbError,
+    DbError(DbError),
+    MemoryAllocationError,
+    MemoryReadError,
+    MemoryWriteError,
+    InputInvalid,
+    InputEmpty,
+    InputWrongPrefix,
+    InputWrongLength,
+    OutputWrongLength,
+    NonExistentImportFunction,
```

Comment:
please move these variants to their own enum like `DbError`, called `CommunicationError` or `ContractAbiError`

Reply:
Why?? Seems to me like wrapping for the sake of wrapping

Reply:
well, sort of.
it's mostly to keep things organized, much like how DbError was implemented.